pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1186,6 +1186,20 @@ self: super: {
     '';
   }) super.cryptol;
 
+  inherit (
+    let
+      fixZ3 = z3: overrideCabal (drv: {
+        preBuild = drv.preBuild or "" + ''
+          # These constants were removed in z3 4.11.
+          export NIX_CFLAGS_COMPILE="-DZ3_bool=bool -DZ3_TRUE=true -DZ3_FALSE=false"
```

Comment:
It's better to use e.g. `appendConfigureFlag` and pass this via Cabal and GHC:

```nix
appendConfigureFlags [ "--ghc-option=-optc=-DZ3_bool=bool" "--ghc-option=-optc=…" … ]
```

https://downloads.haskell.org/ghc/9.10.1/docs/users_guide/phases.html?highlight=optc#ghc-flag--optc%20%E2%9F%A8option%E2%9F%A9

Reply:
I've tried this:

```nix
  inherit (
    let
      fixZ3 = appendConfigureFlags [
        "--ghc-option=-DZ3_bool=bool"
        "--ghc-option=-DZ3_TRUE=true"
        "--ghc-option=-DZ3_FALSE=false"
      ];
    in {
      hz3 = fixZ3 super.hz3;
      z3 = fixZ3 super.z3;
    }
  ) hz3 z3;
```

(c.f. https://downloads.haskell.org/ghc/9.6.6/docs/users_guide/phases.html)

Whether it's `--ghc-option=-DZ3_bool=bool` or `--ghc-option=-optc=-DZ3_bool=bool` or `--ghc-option=-optc-DZ3_bool=bool` or `--ghc-option=-optP-DZ3_bool=bool` with `--ghc-option=-cpp` I can't get it to build :-(

Reply:
Should be possibly somehow. In any case, seems easiest just to apply the upstream patch for this: https://github.com/IagoAbal/haskell-z3/pull/88

Reply:
Good call, did that. hz3 is unmaintained, so put it back on z3 4.8.